src/checker/check.go

```diff
@@ -0,0 +1,81 @@
+package checker
+
+import (
+	"regexp"
+)
+
+type Rule struct {
+	url          string
+	name         string
+	isInProgress bool
+}
+
+type RulesHistory map[string]Rule
+
+func (r RulesHistory) Contains(url string, name string) bool {
+	_, ok := r[url+name]
+	return ok
+}
+
+func (r RulesHistory) IsRuleInProgress(url string, name string) bool {
+	val := r[url+name]
+	return val.isInProgress
+}
+
+func (r RulesHistory) Push(url string, name string, isInProgress bool) {
+	r[url+name] = Rule{
+		url:          url,
+		name:         name,
+		isInProgress: isInProgress,
+	}
+}
+
+func (r RulesHistory) SetProgress(url string, name string, isInProgress bool) {
+	rule := r[url+name]
+	rule.isInProgress = isInProgress
+}
+
+func CheckRuleScript(script string) (bool, error) {
+ 	rulesHistory := &RulesHistory{}
+	return ExecuteLuaMain(script, rulesHistory)
+}
+
+func CheckRule(rulesHistory *RulesHistory, url string, name string) bool {
```

Comment:
```go
func CheckRule(rulesHistory *RulesHistory, url string, name string) (bool, err) {
```

Reply:
@cooligus when you write code in a review please add it as suggestion
![image](https://github.com/Nasz-Elektryk/spito-rules/assets/94986692/eb85518b-6bd3-4f75-9844-6a9befc2d29e)

Reply:
ok